packages/core/src/manifest/ManifestYaml.tsx

```diff
@@ -1,26 +1,42 @@
 import React from 'react';
 import { Modal } from 'react-bootstrap';
+import { ArtifactService } from '../pipeline/config/triggers/artifacts/ArtifactService';
+import { decodeUnicodeBase64 } from '../utils'; 
 
-export interface IManifestYamlProps {
+type IManifestYamlProps = {
   linkName: string;
-  manifestText: string;
   modalTitle: string;
-}
+} & ({ manifestText: string; manifestUri?: never } | { manifestText?: never; manifestUri: string });
 
-export function ManifestYaml(props: IManifestYamlProps) {
+export function ManifestYaml({ linkName, modalTitle, manifestText, manifestUri }: IManifestYamlProps) {
   const [modalVisible, setModalVisible] = React.useState<boolean>(false);
   const toggle = () => setModalVisible(!modalVisible);
+  const [fetchedManifestText, setFetchedManifestText] = React.useState<string>('Loading...');
+
+  React.useEffect(() => {
+    if (manifestUri) {
+      ArtifactService.getArtifactByContentReference(manifestUri)
+        .then((manifest) => setFetchedManifestText(decodeUnicodeBase64(manifest.reference)))
+        .catch((e) => setFetchedManifestText(`Error:  ${typeof e !== 'string' ? JSON.stringify(e) : e}`));
```

Comment:
if this happens the result is kind of weird 
![image](https://github.com/spinnaker/deck/assets/4973876/e62eac3f-9cde-435d-9208-514b33ab7d9b)
should we use another component to show the error? instead of using the same `ManifestYaml`

Reply:
Okay, fixed that. Looks like this now
![Screen Shot 2023-07-06 at 5 30 09 PM](https://github.com/spinnaker/deck/assets/1964445/684d1185-60bc-4a00-86b2-5f2f801d7647)